### PR TITLE
REWORK 0.12

### PR DESCRIPTION
**Pull request by riblanc**
Reworking the whole project to fix errors introduced with zig 0.12.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
       - if: steps.cache-zig-installation.outputs.cache-hit != 'true'
         name: Install zig
         run: |
-          wget https://ziglang.org/download/0.11.0/zig-linux-x86_64-${ZIG_VERSION}.tar.xz -qO /tmp/zig.tar.xz
+          wget https://ziglang.org/download/0.12.0/zig-linux-x86_64-${ZIG_VERSION}.tar.xz -qO /tmp/zig.tar.xz
```
Comment (jgiron42) [inline .github/workflows/ci.yml:37]: why not reuse the ZIG_VERSION variable?

File: src/boot.zig [hunk 2]
```diff
@@ -21,7 +21,7 @@ pub const ft_options: @import("ft/ft.zig").Options = .{
     .logFn = @import("logger.zig").kernel_log,
 };
 
-export fn _entry() linksection(".bootstrap_code") callconv(.Naked) noreturn {
+export fn _start() linksection(".bootstrap_code") callconv(.Naked) noreturn {
```
Comment (jgiron42) [inline src/boot.zig:24]: why change the name?

Comment (riblanc) [reply src/boot.zig:24]: For any reason, ld seems broken on zig 0.12:

image

File: .github/workflows/ci.yml (hunk 1 above)
Comment (riblanc) [reply .github/workflows/ci.yml]: Done

File: src/boot.zig (hunk 2 above)
Comment (riblanc) [reply src/boot.zig:24]: https://github.com/ziglang/zig/issues/18482#issuecomment-1881752824

Comment (riblanc) [reply src/boot.zig]: I finally went back to ``_entry``, now defining the name of the entry symbol in the build script.

**Review by jgiron42 — APPROVED** [review]

**Pull request merged**